channeldb/invoices.go

```diff
@@ -166,6 +166,13 @@ func (c ContractState) String() string {
 // the necessary conditions required before the invoice can be considered fully
 // settled by the payee.
 type ContractTerm struct {
+	// FinalCltvDelta is the minimum required number of blocks before htlc
+	// expiry when the invoice is accepted.
+	FinalCltvDelta int32
```

Comment:
Should we use the opportunity to remove the `ContractTerm` struct? Doesn't seem it add much, it only adds to the confusion where to put the fields. 

As it is not serialized separately either, I would prefer having all these fields on the invoice instead.

Reply:
I'd be down with removing it, it was created back in a time when the realm of possibilities was a bit more unbounded...🌈

Reply:
imo like having a separation of invoice parameters and invoice metadata. the latest restructuring moves the immutable constraints into `ContractTerm`, which are ultimately embedded in the payment request. things like `State`, which is a) mutable, and b) not present in the payreq, is now contained on the main body with the other metadata.

if we don't think this adds any value tho we can flatten entirely.

Reply:
ultimately tho this doesn't affect the migration at all, since we have a working copy of the serialization in `channeldb` and are free to change the in-memory structure until the next one

Reply:
If we are to keep the `ContractTerms`, I think there are a few more immutable things to move there? pay req, memo perhaps. I would still prefer just flatten it, but not a blocker  :)